src/System.Net.Http/src/System/Net/Http/ByteArrayContent.cs

```diff
@@ -62,9 +62,13 @@ protected internal override bool TryComputeLength(out long length)
             return true;
         }
 
-        protected override Task<Stream> CreateContentReadStreamAsync()
-        {
-            return Task.FromResult<Stream>(new MemoryStream(_content, _offset, _count, writable: false));
-        }
+        protected override Task<Stream> CreateContentReadStreamAsync() =>
+            Task.FromResult<Stream>(CreateMemoryStreamForByteArray());
+
+        internal override ValueTask<Stream> CreateContentReadStreamValueAsync() =>
+            GetType() == typeof(ByteArrayContent) ? new ValueTask<Stream>(CreateMemoryStreamForByteArray()) :
```

Comment:
Derived types may override CreateContentReadStreamAsync. If they do and we don't delegate to it, callers of this method will get the wrong behavior. As there's no fast way to ask "does this instance's method X == this type's method X", the best we can do is base it on the type itself. We could implement expensive reflection-based logic and cache the results, but that's unlikely to be worth it. If the runtime provided such functionality, we could switch to that in the future.